js/newTab.js

```diff
@@ -119,6 +119,10 @@
   // Document is loaded. Initialize the New Tab Page.
   gNewTab.init();
   document.addEventListener("NewTabCommandReady", async(function* () {
+    let sw = (yield navigator.serviceWorker.ready).active;
```

Comment:
I'm wondering... could we just make DirectoryLinksProvider initialize when the service worker becomes active (i.e., in sw.js)? That way, we don't need to send this message, or is this message needed?